testdata/src/go.uber.org/errormessage/errormessage.go

```diff
@@ -0,0 +1,96 @@
+//  Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This package tests _single_ package inference. Due to limitations of `analysistest` framework,
+// multi-package inference is tested by our integration test suites. Please see
+// `testdata/README.md` for more details.
+
+package errormessage
+
+var dummy bool
+
+func test1(x *int) {
+	x = nil
+	print(*x) //want "`nil` to `x`"
+}
+
+func test2(x *int) {
+	x = nil
```

Comment:
Can we also add cases where `x` is 

(1) read from a map without guarding (default nilable)
(2) reading from a nil channel perhaps?
(3) an empty slice

Reply:
Added.